src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1152,6 +1163,18 @@ else if (!isDay && world.isDaytime())
         workManager.onWorldTick(event);
     }
 
+    /**
+     * Calculate randomly if the colony should update the citizens.
+     * By mean they update it at CITIZEN_CLEANUP_TICK_INCREMENT.
+     *
+     * @param world the world.
+     * @return a boolean by random.
+     */
+    private static boolean shallUpdate(final World world)
+    {
+        return world.getWorldTime() % (new Random().nextInt(CITIZEN_CLEANUP_TICK_INCREMENT*2) + 1) == 0;
```

Comment:
Use a static Random some where. Or the worlds seed Random. Creating a new Random every tick is a big performance hit as it has to be seeded everytime.